### ISSUE TITLE
Sporadic failures when provisioning Sonarqube projects 

### ISSUE THREAD

**Issue by jevjay** [jdamata/terraform-provider-sonarqube#39]
Hi there,

I am experiencing issues with a number of Sonarqube provider-managed projects, where it seems that a Terraform provider fails to process data retrieved from Sonarqube APIs.

When provisioning new project in Sonarqube, I see a following error:

```

Error: Provider produced inconsistent result after apply

When applying changes to module.tf-postfix-repo.sonarqube_project.repo,
provider "registry.terraform.io/jdamata/sonarqube" produced an unexpected new
value: Root resource was present, but now absent.

 This is a bug in the provider, which should be reported in the provider's own
issue tracker.

```

This error does not prevent Sonarqube providers from provisioning a project in Sonarqube, but I believe it fails to import/process data to store it in a state file. For example, if I am trying to import created resource I get:

```

Error: nil entry in ImportState results. This is always a bug with
the resource that is being imported. Please report this as
a bug to Terraform.

```

The project name I am importing: `tf-postfix`

_Note: it is worth mentioning that we observed this issue only for the projects which start with `t` character_

### Terraform Version
0.13.5

### Affected Resource(s)
Please list the resources as a list, for example:
- sonarqube_project

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
```hcl
# Sonarqube project configuration
#
# This file contains all configurations for a sonarqube project, which is used for multiple code validations

resource "sonarqube_project" "repo" {
  name       = var.name
  project    = var.name
  visibility = "public"
}

# Associate a default PR validation quality gate to a Sonarqube project
resource "sonarqube_qualitygate_project_association" "default_pr" {
  gatename   = local.default_pr_validation_qg
  projectkey = sonarqube_project.repo.project
}

# Associate custom quality gates to the project. 
# Configuration taken from `custom_quality_gates` map
resource "sonarqube_qualitygate" "custom" {
  for_each = var.custom_quality_gates

  name = each.key
}

resource "sonarqube_qualitygate_project_association" "custom" {
  for_each = var.custom_quality_gates

  gatename   = sonarqube_qualitygate.custom[each.key].id
  projectkey = sonarqube_project.repo.project
}

resource "sonarqube_qualitygate_condition" "custom" {
  for_each = var.custom_quality_gates

  gatename  = sonarqube_qualitygate.custom[each.key].id
  metric    = each.value.metric
  threshold = each.value.threshold
  op        = each.value.operator
}
```

_Note: the Sonarqube resources provisioned as a separate internal Terraform module_

### Expected Behavior
Terraform should be able to provision/import Sonarqube projects

### Actual Behavior
Terraform failed to import a Sonarqube project into a state file, thus resulting in an untracked resource

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

### Important Factoids
Sonarqube version: 8.9
Sonarqube provider version: 0.0.3

**Comment by jdamata** [comment]
I'm having a little trouble reproducing this. I added an acceptance test that creates and imports a project with the name tf-postfix and it worked ok.

Commit: https://github.com/jdamata/terraform-provider-sonarqube/commit/346516e7068a84af3bda4f67e3543d2920be515a
Build: https://cloud.drone.io/jdamata/terraform-provider-sonarqube/75/2/4

As a quick check, can you try something like:

```hcl
resource "sonarqube_project" "tf-postfix-test" {
  name       = "tf-postfix-test"
  project    = "tf-postfix-test"
  visibility = "public"
}
```

Then do:
```sh 
terraform state rm 'sonarqube_project.tf-postfix-test'
terraform import 'sonarqube_project.tf-postfix-test' tf-postfix-test
```

Just want to try and narrow down whats causing this bug.

**Comment by jevjay** [comment]
When trying to run:

```
terraform state rm 'sonarqube_project.tf-postfix-test'
```

here is the response I get:

```
Error: Invalid target address

No matching objects found. To view the available instances, use "terraform
state list". Please modify the address to reference a specific instance.

Releasing state lock. This may take a few moments...
```

and when I try to import it I get:

```
module.tf-postfix-repo.sonarqube_project.repo: Importing from ID "tf-postfix"...

Error: nil entry in ImportState results. This is always a bug with
the resource that is being imported. Please report this as
a bug to Terraform.


Releasing state lock. This may take a few moments...
```

Here is a TRACE level logging for the Terraform state:

```
2021/06/22 14:08:14 [TRACE] [walkImport] Entering eval tree: module.tf-postfix-repo.sonarqube_project.repo (import id "tf-postfix")
2021/06/22 14:08:14 [TRACE] eval: *terraform.EvalSequence
2021/06/22 14:08:14 [TRACE] eval: *terraform.EvalGetProvider
2021/06/22 14:08:14 [TRACE] eval: *terraform.EvalImportState
2021/06/22 14:08:14 [TRACE] GRPCProvider: ImportResourceState
module.tf-postfix-repo.sonarqube_project.repo: Importing from ID "tf-postfix"...
2021-06-22T14:08:14.346+0200 [DEBUG] plugin.terraform-provider-sonarqube_v0.0.3: 2021/06/22 14:08:14 [DEBUG] GET http://***:***@rev-sonarqube.k8s.prod.revinate.net/api/projects/search?project=tf-postfix
2021/06/22 14:08:14 [ERROR] eval: *terraform.EvalImportState, err: nil entry in ImportState results. This is always a bug with
the resource that is being imported. Please report this as
a bug to Terraform.
2021/06/22 14:08:14 [ERROR] eval: *terraform.EvalSequence, err: nil entry in ImportState results. This is always a bug with
the resource that is being imported. Please report this as
a bug to Terraform.
2021/06/22 14:08:14 [TRACE] [walkImport] Exiting eval tree: module.tf-postfix-repo.sonarqube_project.repo (import id "tf-postfix")
2021/06/22 14:08:14 [TRACE] vertex "module.tf-postfix-repo.sonarqube_project.repo (import id \"tf-postfix\")": visit complete
2021/06/22 14:08:14 [TRACE] dag/walk: upstream of "module.tf-postfix-repo (close)" errored, so skipping
2021/06/22 14:08:14 [TRACE] dag/walk: upstream of "provider[\"registry.terraform.io/jdamata/sonarqube\"] (close)" errored, so skipping
2021/06/22 14:08:14 [TRACE] dag/walk: upstream of "root" errored, so skipping

Error: nil entry in ImportState results. This is always a bug with
the resource that is being imported. Please report this as
a bug to Terraform.


2021/06/22 14:08:14 [DEBUG] [aws-sdk-go] DEBUG: Request dynamodb/GetItem Details:
```

when I have tried to follow the API search URL (which Sonarqube provider tries to access) I was able to see the output from APIs thus it seems to me that somehow provider is not able to parse the API output (somehow)

As I am using sonarqube provider v0.0.3 do you think I shall upgrade to v0.0.5?

**Comment by jdamata** [comment]
I've pushed out version v0.0.6 that contains some fixes for imports related to sonarqube_qualitygate_project_association and sonarqube_qualityprofile_project_association. Please try this version.

Also I added some examples that seem to work for me: https://github.com/jdamata/terraform-provider-sonarqube/tree/master/examples

If version v0.0.6 does not work for you, can you try using those examples with your version of tf/sonarqube and see if those examples work.
